### DISCUSSION TITLE
Files.app and local download

### DISCUSSION THREAD

**Discussion by srl295** [blinksh/blink#1601] [Support] [answered]
was testing out Files.app, pretty useful!

I noticed there's a 'download now' option in File.app which seems to work, however, 
1. the files are not accessible offline as far as i could tell
2. there does not seem to be an option to remove the cached files.

**Comment by carloscabanero** [comment, marked as answer]
1. This is by design. One of the assumptions when using Blink is that you are always connected. When designing how far we wanted to take our integration, we just wanted this to be a better solution than the "interactive" sftp command shell. We also provide a sftp --update that will keep files in sync between two locations.
2. The cache is stored in an upper directory, File Provider Storage. You can remove the cache manually, doing `rm -rf ../File\ Provider\ Storage`

All in all, we think the Files.app integration has seen very little use, most people prefer the commands. Despite this, it may become more useful for Blink the future and we will update to the new API. So please, let us know what you would use Offline access for and we will take it into account.

**Reply by srl295** [reply]
i mostly use it for moving files from a server to icloud and back. I guess I could copy files to the 'local' directory in files.app itself.

**Comment by aikinai** [comment]
I love the Files integration since I have a lot of files on a Mac sitting at home and not necessarily in iCloud. Of course if I'm actively using Blink, I'll use the command line, but I took the Files app integration as just a bonus feature basically unrelated to usage of the Blink shell itself. It's just a great way to have access to my "local" files from anywhere.

I'm the one that posted in #1619 about deleted the app to clear the cache. I see now I can do it through the shell, so I'm personally okay, but there's no way anyone will know that without finding this issue. Would it be possible to expose the cache in the Documents section under the app in Storage settings? I see some other apps do that, so you can see what's taking up storage and delete easily from there.

**Reply by carloscabanero** [reply]
This is a good solution. Will see what we can do.

**Reply by carloscabanero** [reply]
From 16.0.4, the file cache should now be deleted when you remove the Files.app directory in the application.

This is not yet ideal, and we will have to move to the new API at one point. But hope this two options help while we do that.

**Comment by carloscabanero** [comment]
From 16.1.9, we are now also trying to skip the File Provider Storage from the iCloud backup.

I created https://github.com/blinksh/blink/issues/1789 to track the specifics now during testing.
